tmpdir.go

```diff
@@ -26,3 +28,69 @@ func Setup(path string) (err error) {
 func Cleanup() error {
 	return os.RemoveAll(Dir)
 }
+
+// MkTemp creates a unique filename in the temporary directory and returns a
+// writable file descriptor. This function is safe for any arbitrary filename.
+//
+// Subdirectories are not supported. The Setup() function must be called first.
+func MkTemp(filename string) (*os.File, error) {
+	if Dir == "" {
```

Comment:
Any good reason to use the global namespace? I've seen this as a common practice but introduced a few issues:
1. Test cases may interfere with each other
2. Read/write within goroutines is not safe
3. Nobody has control over it.
4. Cannot reuse the package at runtime(safely anyway)
5. package is `broken` from the get-go

If there's an actual need for this, we may as well look into a singleton(https://golang.org/pkg/sync/#Once.Do)

Reply:
The primary use case of this package is to load at startup and then write files to it.  So it wouldn't be initialized more than once.  But I do agree it probably shouldn't be public or should be protected with sync.

Reply:
The idea is that there's one global "temp directory". The reason is like this is because ... this is how it was written in Desk :-)

Since we didn't have any helper functions before this PR, exposing `Dir` was needed. I'm not against changing that, for example by unexporting `Dir`, but doing it is a bit outside the scope of this PR, I think.

Reply:
I'll look at this after this PR is merged btw